spring-vault-core/src/test/java/org/springframework/vault/core/VaultKeyValueTemplateVersionedIntegrationTests.java

```diff
@@ -35,4 +45,28 @@
 		super("versioned", KeyValueBackend.versioned());
 	}
 
+	@Test
+	void shouldPatchSecret() {
+		final String oldKey = "key";
+		final String newKey = "newKey";
+		Map<String, String> secret = Collections.singletonMap(oldKey, "value");
+
+		String key = UUID.randomUUID().toString();
+
+		this.kvOperations.put(key, secret);
+
+		Map<String, String> newSecret = Collections.singletonMap(newKey, "newValue");
+
+		((VaultKeyValue2Template) this.kvOperations).patch(key, newSecret);
```

Comment:
Once the `patch` method gets exposed through `VaultKeyValueOperations`, there's no need to have the downcast.

Reply:
Now it does not downcast: https://github.com/spring-projects/spring-vault/pull/587/commits/95d25bd9120c340cecc2eab71c67ffc047e41f07#diff-98a440907d66c701b5a12de4a5db680aR61